types/throttleit/index.d.ts

```diff
@@ -0,0 +1,18 @@
+// Type definitions for throttleit 1.0
+// Project: https://github.com/component/throttle
+// Definitions by: Ifiok Jr. <https://github.com/ifiokjr>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+type AnyFunction<R = any> = (...args: any[]) => R;
+
+/**
+ * Returns a new function that, when invoked, invokes `func` at most once per `wait` milliseconds.
+ *
+ * @param func Function to wrap.
+ * @param wait Number of milliseconds that must elapse between `func` invocations.
+ * @return A new function that wraps the `func` function passed in.
+ */
+declare function throttle<GFunction extends AnyFunction>(fn: GFunction, wait: number): GFunction;
```

Comment:
You can use the type `Function` instead of `AnyFunction`.

Reply:
The reason I didn't do that was because of this lint error.

> Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void` - `.tslint(ban-types)`

Should I disable that rule in the `tslint.json` file

Reply:
Ah my bad. I discussed about this with the team and we agree is better to avoid `Function` and use `AnyFunction` as you originally posted as `Function` comes with another set of issues. 

I'll merge this. Thanks for contributing.